warehouse/query-core/src/main/java/datawave/mr/bulk/RecordIterator.java

```diff
@@ -530,7 +530,7 @@ protected synchronized void fail(Throwable t) {
             // do nothing, but clean up resources anyway.
         }
         try {
-            Thread.sleep(failureSleep);
+            t.wait(failureSleep);
```

Comment:
If we call wait on the RecordIterator object, this will allow other threads to enter the fail method while we wait.  Calling wait on the throwable won't do that.  Calling Thread.sleep doesn't do that either.  Are we sure this is the behavior we want?

Reply:
Changed the wait(), should be better.